homeassistant/components/xiaomi_miio/vacuum.py

```diff
@@ -187,9 +189,23 @@ async def async_setup_platform(hass, config, async_add_entities, discovery_info=
 
     # Create handler
     _LOGGER.info("Initializing with host %s (token %s...)", host, token[:5])
-    vacuum = Vacuum(host, token)
 
-    mirobo = MiroboVacuum(name, vacuum)
+    unique_id = None
+
+    try:
+        vacuum = Vacuum(host, token)
+        device_info = await hass.async_add_executor_job(vacuum.info)
+        unique_id = f"{device_info.model}-{device_info.mac_address}"
```

Comment:
The `mac` address should be unique across models so you should be able to use it directly.
```suggestion
        unique_id = device_info.mac_address
```

Reply:
I'm OK with this, but was trying to match the rest of the Miio integration which uses the model-unique_id. Is this the preferred method?

Reply:
If there is only one entity using it then the mac address is fine on its own.  I usually try to avoid model strings as vendors sometimes reformat them later if they change product lines.

https://developers.home-assistant.io/docs/entity_registry_index#unique-id-requirements